plugins/protocolbuffers/cpp/v25.1/Dockerfile

```diff
@@ -16,7 +16,9 @@ RUN curl -fsSL -o protoc.tar.gz https://github.com/protocolbuffers/protobuf/rele
  && rm protoc.tar.gz
 RUN bazel build '//:protoc_lib'
 COPY --link BUILD cpp.cc plugins/
-RUN bazel build '//plugins:protoc-gen-cpp.stripped'
+COPY --link disable_cpp_editions.patch .
+RUN git apply < disable_cpp_editions.patch \
+ && bazel build '//plugins:protoc-gen-cpp.stripped'
```

Comment:
If we're updating this plugin lets merge the latest changes from 25.3 or 28.3 into here (including changing base image to bookworm, updating base images to latest, and reducing image layers).